File: Backend/api/schemas/smartscore.py
from __future__ import annotations

import datetime as dt
from typing import Any

from pydantic import BaseModel

from Backend.core.enums import SmartScoreBand


class SmartScoreRead(BaseModel):
    id: str
    score: int
    band: SmartScoreBand
    summary: str
    drivers: dict[str, Any]
    created_at: dt.datetime

    class Config:
        from_attributes = True

File: mobile/lib/data/models/smart_score.dart
class SmartScoreSnapshot {
  SmartScoreSnapshot({
    required this.id,
    required this.score,
    required this.band,
    required this.summary,
    required this.createdAt,
    this.drivers,
  });

  final String id;
  final int score;
  final String band;
  final String summary;
  final DateTime createdAt;
  final Map<String, dynamic>? drivers;

  factory SmartScoreSnapshot.fromJson(Map<String, dynamic> json) {
    return SmartScoreSnapshot(
      id: json['id'] as String,
      score: json['score'] as int,
      band: (json['band'] as String?) ?? '',
      summary: json['summary'] as String? ?? '',
      createdAt: DateTime.parse(json['created_at'] as String),
      drivers: (json['drivers'] as Map<String, dynamic>?),
    );
  }
}
